domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterControllerDebugImplTest.kt

```diff
@@ -813,6 +813,118 @@ class PlatformParameterControllerDebugImplTest {
     assertThat(updatedParam?.syncStatus).isEqualTo(SyncStatus.LOCAL_OVERRIDE)
   }
 
+  @Test
+  fun testResetPlatformParameter_withLocalOverride_resetsParameter_returnsDefaultSyncStatus() {
+    executeInPreviousAppInstance { testComponent ->
+      addTestIntegerOverriddenPlatformParameterToDatabase(
+        testComponent,
+        TEST_LOCAL_OVERRIDE_SYNC_UP_WORKER_PERIOD_HOURS
+      )
+      testComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    val resetProvider = platformParameterControllerDebugImpl.resetPlatformParameters(
+      listOf(PlatformParameterId.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS)
+    )
+    monitorFactory.waitForNextSuccessfulResult(resetProvider)
+    val ephemeralPlatformParametersProvider =
+      platformParameterControllerDebugImpl.loadEphemeralPlatformParameters()
+    val ephemeralPlatformParameters =
+      monitorFactory.waitForNextSuccessfulResult(ephemeralPlatformParametersProvider)
+    val ephemeralSyncUpWorkerTimePeriodValue = ephemeralPlatformParameters
+      .find { it.id == PlatformParameterId.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS }
+
+    assertThat(ephemeralSyncUpWorkerTimePeriodValue?.syncStatus)
+      .isEqualTo(SyncStatus.NOT_SYNCED_FROM_SERVER)
+  }
+
+  @Test
+  fun testResetPlatformParameter_withRemoteAndLocalOverride_resetsParameter_returnsServerStatus() {
+    executeInPreviousAppInstance { testComponent ->
+      addTestIntegerRemotePlatformParameterToDatabase(
+        testComponent,
+        TEST_REMOTE_SYNC_UP_WORKER_PERIOD_HOURS
+      )
+      addTestIntegerOverriddenPlatformParameterToDatabase(
+        testComponent,
+        TEST_LOCAL_OVERRIDE_SYNC_UP_WORKER_PERIOD_HOURS
+      )
+      testComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    val resetProvider = platformParameterControllerDebugImpl.resetPlatformParameters(
+      listOf(PlatformParameterId.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS)
+    )
+    monitorFactory.waitForNextSuccessfulResult(resetProvider)
+    val ephemeralPlatformParametersProvider =
+      platformParameterControllerDebugImpl.loadEphemeralPlatformParameters()
+    val ephemeralPlatformParameters =
+      monitorFactory.waitForNextSuccessfulResult(ephemeralPlatformParametersProvider)
+    val ephemeralSyncUpWorkerTimePeriodValue = ephemeralPlatformParameters
+      .find { it.id == PlatformParameterId.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS }
+
+    assertThat(ephemeralSyncUpWorkerTimePeriodValue?.syncStatus)
+      .isEqualTo(SyncStatus.SYNCED_FROM_SERVER)
+  }
+
+  @Test
+  fun testResetFeatureFlag_withLocalOverride_resetsFlag_returnsDefaultSyncStatus() {
```

Comment:
Here and elsewhere,
1. testResetFeatureFlag(s) as named in the API.
2. _NoRemote -> to denote fallback is default.
```suggestion
  fun testResetFeatureFlags_withLocalOverrideAndNoRemote_resetsFlag_returnsDefaultSyncStatus() {
```

Reply:
updated